src/detect.ts

```diff
@@ -15,27 +15,66 @@ export function getCurrentBody(): string {
   return github.context.payload.pull_request.body || ''
 }
 
+const CHECK_REGEXP = /^\s*- \[x\] /
+const UNCHECK_REGEXP = /^\s*- \[ \] /
+
+export function getPreviousChecked(previousBody: string): string[] {
+  const previousLines = previousBody.split('\n')
+  return previousLines
+    .filter(line => CHECK_REGEXP.test(line))
+    .map(line =>
+      line
+        .replace(CHECK_REGEXP, '')
+        // remove trailing newline
+        .replace(/[\n\r]/, '')
+    )
+}
+
+export function getPreviousUnchecked(previousBody: string): string[] {
+  const previousLines = previousBody.split('\n')
+  return previousLines
+    .filter(line => UNCHECK_REGEXP.test(line))
+    .map(line =>
+      line
+        .replace(UNCHECK_REGEXP, '')
+        // remove trailing newline
+        .replace(/[\n\r]/, '')
+    )
+}
+
+export function getCurrentChecked(currentBody: string): string[] {
+  const currentLines = currentBody.split('\n')
+  return currentLines
+    .filter(line => CHECK_REGEXP.test(line))
+    .map(line =>
+      line
+        .replace(CHECK_REGEXP, '')
+        // remove trailing newline
+        .replace(/[\n\r]/, '')
+    )
+}
+
+export function getCurrentUnchecked(currentBody: string): string[] {
+  const currentLines = currentBody.split('\n')
```

Comment:
I am concerned that having currentLines split occur twice may slow down the speed when the text becomes long.

Reply:
yep same thoughts, I've got local changes where the input changes from `previousBody: string` to `previousLines: string[]`